airflow/www/templates/airflow/dag.html

```diff
@@ -290,6 +294,7 @@ <h4 class="modal-title" id="dagModalLabel">
 {% endblock %}
 {% block tail %}
   {{ super() }}
+  <script src="{{ url_for_asset('utils.js') }}"></script>
```

Comment:
It seems most/all pages include this file. Should we either add it in to base.js, or add this file in baselayout instead?

Reply:
Currently, `tail` and` tail_js` blocks are not used correctly in our application. So the introduction of your suggestion can be problematic, because it is difficult to predict the order in which items are loaded.

Reply:
I would prefer to deal with this problem in a separate PR. When the JS code will be in separate files and it will be possible to introduce mechanisms that will be much more efficient in loading the JS code. Currently, it is not possible to implement a solution that will solve the problem completely and I will be 100% sure. I can make another change depending on this in seperate PR, but it creates another chain of PR.  Already in the case of CSS code, we are blocked from making changes, because the introduction of new changes involves the necessity to fix the old problems first.

I could introduce your solution now, but it will complicate the code, which is not in good condition.

It is worth noting that if a reference to the same JS file appears in the HTML code, the file is only loaded once. This is a simple and easy module loader.  This is not a bad thing if you see the dependency twice in the HTML code. This means that a given fragment is needed in two places. The browser will provide an automatic and 100% sure way that the code will not be executed twice. This is not a bad practice, but a simple way to ensure that the module is loaded.

Reply:
I guess you would like me to apply a solution similar to that presented in the PR #4950 but this is not in my opinion the right solution. This creates one big file that contains the major part of JS code. I would like to divide the code into smaller parts and mark the requirements for each JS snippet to make future refactorations easier.

Reply:
Alright, sounds good as it is.